configdialog.cpp

```diff
@@ -92,8 +111,23 @@ void ConfigDialog::refresh() {
 		SendMessage(slider, TBM_SETPOS, TRUE, encodef(m_current_config.at(entry.first)));
 	}
 
+	if (m_current_config.at(YonkPattern) < 0 || m_current_config.at(YonkPattern) >= _PATTERN_COUNT)
```

Comment:
OK, here are some thoughts. TL;DR, I don't think we should show a message box in these cases, we should just silently fix the value and move on.

First of all, it feels like this validation + repair stuff really wants to be a separate system in the code. It might be as simple as just having a function `is_config_value_in_range` that we can call rather than doing these if checks manually. This is especially true if we're going to throw up a message box on a validation failure, since I do not want ugly message box function calls in our regular app logic. I want to think about this some more though, because I'm not 100% sure how we should design it, so I'll make another issue for it and we can address it later.

Second, the message boxes themselves. We need to be extremely reverent of users' time and attention and only pester them with a message box when it is absolutely necessary, and if we do pester them, make our messages _extremely_ concise, easy to understand, and crystal-clear what, if anything, is required of them. These kinds of decisions, more than any technical argument about architecture or naming or whatever, are what separate bad software from good software, and good software from great software. 

That being said, deciding what this message box should say, and deciding if it should show up _at all_, is an extremely hard problem, and I don't have an answer for it right now. However, my strong inclination is that we should not show a message box in the first place. There are a vast number of different reasons why this validation failure might happen, from bugs in our code to users manually editing their registry to downgrading to an old version of the screensaver. In all of these cases, the important thing is our shit needs to just keep working. There is very we or the user can do if a config value gets corrupted, besides just set it back to a known good value. Plus, this is not banking information or passwords - these settings are inherently visual (because they can easily be spotted and adjusted back to preferred settings), not necessarily permanent (because people might be changing their config often), and not important to people's lives (it's a got damn screensaver). So, do we really need to annoy the user with a message box and take a few seconds out of their day? My instinct is no.